packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,542 @@
+  late final FlutterView _view;
+}
+
+/// Delegate class for regular window controller.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+mixin class RegularWindowControllerDelegate {
+  /// Invoked when user attempts to close the window. Default implementation
+  /// destroys the window. Subclass can override the behavior to delay
+  /// or prevent the window from closing.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void onWindowCloseRequested(RegularWindowController controller) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+
+    controller.destroy();
+  }
+
+  /// Invoked when the window is closed.
+  ///
+  /// Default implementation exits the application if this was
+  /// the last top-level window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void onWindowDestroyed() {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    if (!owner.hasTopLevelWindows()) {
+      // TODO(mattkae): close the application if this is the last window
+      // via ServicesBinding.instance.exitApplication(AppExitType.cancelable);
+    }
+  }
+}
+
+/// A controller for a regular window.
+///
+/// A regular window is a traditional window that can be resized, minimized,
+/// maximized, and closed. Upon construction, the window is created for the
+/// platform with the provided properties.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [RegularWindow] widget, who does the work of rendering the
+/// content inside of this window.
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   contentSize: const WindowSizing(
+///     preferredSize: Size(800, 600),
+///     preferredConstraints: BoxConstraints(minWidth: 640, minHeight: 480),
+///   ),
+///   title: "Example Window",
+/// );
+/// runWidget(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// When provided to a [RegularWindow] widget, widgets inside of the [child]
+/// parameter will have access to the [RegularWindowController] via the
+/// [WindowControllerContext] widget.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+abstract class RegularWindowController extends BaseWindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  ///
+  /// Upon construction, the window is created for the platform.
+  ///
+  /// [contentSize] sizing requests for the window. This might not behonored by the platform
+  /// [title] the title of the window
+  /// [state] the initial state of the window
+  /// [delegate] optional delegate for the controller controller.
```

Comment:
Generally we use more prose-like docs. Maybe something like:

```suggestion
  /// The [contentSize] argument sets the window's initial size preference.
  /// This might not be honored by the platform.
  ///
  /// The `title` argument configures the window's initial title.
  /// If omitted, some platforms might fall back to the app's name.
  ///
  /// The `delegate` argument can be used to listen to the window's
  /// lifecycle. For example, it can be used to save state before
  /// a window is closed.
```